### ISSUE TITLE
URLs to permalinks within collapsed admonition blocks don't scroll to or expand the block upon initial page load

### ISSUE THREAD

**Issue by kleinen** [squidfunk/mkdocs-material#3032]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

The initial loading of a URL to a permalink located within a collapsible admonition block will not scroll to that link nor automatically expand the collapsed block.  This used to work in past versions of mkdocs-material, but I am not sure exactly what version this changed in.

### Expected behaviour

A URL to a permalink located within a collapsible admonition block, when loaded in a new browser window or tab, would automatically scroll to that block and expand it to where the permalink is located.

### Actual behaviour

The page loads, but it loads as if no permalink was used.  Also, using the same permalink from the TOC does not change the page (since it must think its already at that page).  Switching to a different TOC link on the same page and clicking on the original permalink in the TOC works.

### Steps to reproduce

1. Using the included mkdocs.yml (from below) and the following `docs/index.md` document:

```html
# Test

??? success "Section 1"
    ## Section 1

    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do...

??? success "Section 2"
    ## Section 2

    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do...
```

2. Open a new browser window or tab and go to http://0.0.0.0:8000/#section-2 (assuming you are hosting from 0.0.0.0:8000).
3. Notice that it does not automatically scroll to Section 2 or expand the collapsed admonition block.

### Package versions

Using the `squidfunk/mkdocs-material:7.2.6` docker container hosted on Ubuntu 20.04 Linux

### Configuration

```yaml
site_name: 'Site'

markdown_extensions:
  - admonition
  - pymdownx.details
```

### System information

- Operating system: `squidfunk/mkdocs-material` docker container hosted on Ubuntu 20.04 Linux
- Browser: Chrome, Firefox

**Comment by squidfunk** [comment]
Thanks for reporting. What’s the last version that worked for you? Could you try narrowing down the version that broke it?

**Comment by kleinen** [comment]
Sure.  I'll do my best to narrow it down.

**Comment by kleinen** [comment]
I went back and it seems as if it stopped working when the project released 7.0.1 (the docker container tag).

BTW, mad props on how responsive you are. 😄

**Comment by squidfunk** [comment]
Thanks for narrowing it down, I’ll take a look when I find some time!

**Comment by kleinen** [comment]
You are truly appreciated!  Thanks!

**Comment by squidfunk** [comment]
Fixed in 726888cb5. This bug is indeed a regression related to the rewrite in 7.0.0. I'll revisit the whole anchor/hash positioning pipeline at some point in the future, as it needs some refactoring, probably when [anchor tracking](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#anchor-tracking) made its way into master.

**Comment by squidfunk** [comment]
Released as part of 7.2.7.
